port/resource_port_blueprint.go

```diff
@@ -492,16 +545,21 @@ func blueprintResourceToBody(d *schema.ResourceData) (*cli.Blueprint, error) {
 				}
 				propFields.Items = items
 			}
-		case "object":
-			if d, ok := p["default"]; ok && d.(string) != "" {
-				defaultObj := make(map[string]interface{})
-				err := json.Unmarshal([]byte(d.(string)), &defaultObj)
-				if err != nil {
-					return nil, err
+		}
+
+		if defaultOk && df != "" {
+			if deprecatedValue, ok := p["default"]; ok && deprecatedValue != "" {
+				setDefaultToBody(deprecatedValue.(string), &propFields)
+			}
+		} else {
+			if defaultValue, ok := p["default_value"].(map[string]interface{}); ok && len(defaultValue) != 0 {
+				if _, ok := defaultValue["value"]; !ok {
+					return nil, fmt.Errorf("default value for property %s is missing", p["identifier"].(string))
```

Comment:
The error should be more clear like "The default map is missing the value key for property %s"